worktree.c

```diff
@@ -52,7 +52,7 @@ static struct worktree *get_main_worktree(void)
 
 	strbuf_add_absolute_path(&worktree_path, get_git_common_dir());
 	if (!strbuf_strip_suffix(&worktree_path, "/.git"))
-		strbuf_strip_suffix(&worktree_path, "/.");
+		strbuf_strip_suffix(&worktree_path, "/.git/.");
```

Comment:
This change seems to solve the problem related to incorrect worktree path. 
Test `93` & `94` passed successfully that were failing before.

Reply:
But that `"/."` stripping probably was in there for a reason, no? See https://dev.azure.com/gitgitgadget/git/_build/results?buildId=28160&view=ms.vss-test-web.build-test-results-tab&runId=89440&resultId=105546&paneView=debug for two new breakages.

So I think what you'd _rather_ want is something like

```diff
- 	if (!strbuf_strip_suffix(&worktree_path, "/.git"))
+ 	if (!strbuf_strip_suffix(&worktree_path, "/.git") &&
+ 	    !strbuf_strip_suffix(&worktree_path, "/.git/."))
```

On the other hand, this could also simply be achieved via

```diff
- 	if (!strbuf_strip_suffix(&worktree_path, "/.git"))
- 		strbuf_strip_suffix(&worktree_path, "/.");
+ 	strbuf_strip_suffix(&worktree_path, "/.");
+ 	strbuf_strip_suffix(&worktree_path, "/.git");
```

which would be simpler, and my preference.

Reply:
Oh, I totally forgot to congratulate you for finding the exact right spot to fix this!

Reply:
> Oh, I totally forgot to congratulate you for finding the exact right spot to fix this!

Thanks :smile: